external-crates/move/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -140,84 +126,706 @@ struct PackageWithResolverTOML<'a>(&'a Package);
+                for to_pkg_name in g
+                    .package_graph
+                    .neighbors_directed(pkg_name, Direction::Outgoing)
+                {
+                    // unwrap is safe as all edges have a Dependency weight
+                    let sub_dep = g.package_graph.edge_weight(pkg_name, to_pkg_name).unwrap();
+                    self.package_graph
+                        .add_edge(pkg_name, to_pkg_name, sub_dep.clone());
+                }
+            }
+        }
+
+        // insert direct dependency edges and (if necessary) packages for the remaining graph nodes
+        // (not present in package table)
+        for (dep_name, (g, _)) in &dep_graphs {
+            if let Some(dep) = dependencies.get(dep_name) {
+                let internally_resolved =
+                    self.insert_direct_dep(dep, root_package, *dep_name, g, mode, parent)?;
+                // make sure that dependencies of the directly dependent package do not differ from
+                // the dependencies of the same package in other sub-graphs (if any)
+                for (other_dep_name, (other_g, _)) in &dep_graphs {
+                    if dep_name != other_dep_name && other_g.package_graph.contains_node(*dep_name)
+                    {
+                        let (other_pkg_deps, pkg_deps) = deps_equal(*dep_name, &other_g, &g);
+                        if other_pkg_deps != pkg_deps {
+                            bail!(
+                                "When resolving dependencies for package {}, \
+                                         conflicting dependencies found for '{}':\n{}{}",
+                                root_package,
+                                dep_name,
+                                format_deps(*other_dep_name, other_pkg_deps),
+                                format_deps(*dep_name, pkg_deps),
+                            )
+                        }
+                    }
+                }
+
+                if internally_resolved {
+                    // insert edges from the directly dependent package to its neighbors for
+                    // internally resolved sub-graphs - due to how external graphs are constructed,
+                    // edges between directly dependent packages and their neighbors are already in
+                    // the sub-graph and would have been inserted in the first loop in this function
+                    for to_pkg_name in g
+                        .package_graph
+                        .neighbors_directed(*dep_name, Direction::Outgoing)
+                    {
+                        // unwrap is safe as all edges have a Dependency weight
+                        let sub_dep = g.package_graph.edge_weight(*dep_name, to_pkg_name).unwrap();
+                        self.package_graph
+                            .add_edge(*dep_name, to_pkg_name, sub_dep.clone());
+                    }
+                }
+            } else {
+                bail!(
```

Comment:
This error case is getting kind of lost among the other things going on here, can we use `let ... else` to dedent this section and surface it earlier.  I also had a question of when `dep_name` doesn't show up in `dependencies` under normal circumstances, and seeing this error answers that question (it doesn't), so it would be good to surface it next to the check.